### ISSUE TITLE
`vue/html-closing-bracket-spacing` (and more) does not work for top-level tags

### ISSUE THREAD

**Issue by CyanSalt** [vuejs/eslint-plugin-vue#1879]
Currently `vue/html-closing-bracket-spacing` only reports issues in the template, not issues with SFC top-level tags. This behavior is consistent with the implementation, but I'm not sure if this is intentional by design. I would like this rule to support a new option to check and report issues in top-level tags.

**What rule do you want to change?**

[`vue/html-closing-bracket-spacing`](https://eslint.vuejs.org/rules/html-closing-bracket-spacing.html) and [`vue/html-closing-bracket-newline`](https://eslint.vuejs.org/rules/html-closing-bracket-spacing.html). Maybe there are others, I'm not sure.

**Does this change cause the rule to produce more or fewer warnings?**

Yes, it may add warnings. But we can also prevent breaking changes with a new option.

**How will the change be implemented? (New option, new default behavior, etc.)?**

This rule currently contains an object option, so it looks like we just need to add a `checkTopLevelTags: true` option. Further, this option should probably become the default behavior in the next major release.

**Please provide some example code that this change will affect:**

**What does the rule currently do for this code?**

**What will the rule do after it's changed?**

You can compare line 4 and line 16 of this [demo](https://ota-meshi.github.io/eslint-plugin-vue-demo/#eJx1VcuOIzUU/RVTs0gitROJkdCQTjeggQULJCQQG8LCVXWr4mmXXdiudEKUr2DL1/ElHNv1SujZdJfPub6+99xHLllhSsq22c5T0yrh6XmvGdtJ3XaeHXkDVj3ts3j+TaiO9hnbJJu8895o9m2hZPECm4PQpaKP4QSjQgnngIqy5Mlynz1/V5bsV/O92W0SlBx1arT2pjRcSefhIZKglUwfDPFUxsJqGcwemNQlnVb4x8L5R2Tg9tlgu32hc+9wLcsJv3lJ4s7sysBd2GKkOS+NpsU2vrEO3+w6XRmTn9gn9sV4GAyfB/uda4VGHrKa34Ey//7z924TyGd2uaS3PJ08u157FTZKJrU2ncLHbjMr2M4VVraeOfJdGx6TTWusZxdmqXpghWlQPSrZlVXWNGyfHbsYWmG082yqLWLHhSWobDWwo7Q9+fsfoPa66nThJco/q/pyxS4hxPHK+hi8rtvOHZaXlIcst+wn4Q9ri3umWa7W3vzirdT18v1Xq7WDH1ryD6uHZB7U2bJKKEc9ElTZzmJOb0Tyugp/7ynEHRLa62uIOyWVuu+jUcaCHvRZIoGnZ7acCfIN7uZKxI7e4vv1gJ6AOqvHvd5s2Nzbz0ZqT/aHI2nv/uc1mr8RXHhAaumlUP0Tuu8b2IdX0BWxuLHM/qxCvdfTUCXFi5DJFn2VYyaWs+yCB8baFBqnGNtoNx9rBLLo41ggjEWIYoFEH6Nu63Eu03vhi8dotixYzq3C0CSrHLrV1nS65H2A74iqe9s0YJ+98b7Kv/7w/nGW5buq6p1Am16R7CGznSKXbS+huTcQv0GuvJSW0KdHrLgvHyLzyZ1458jxo7BuRJPGPA7QHR1rbLvCG8td15KNKBbR4NzoiNTkg8YWLmxCtOHCnXXBW0yddMTpREUHPwNbCIAlYe1YEfy4kQiLCJedrEdXoZ+EJa6p5n+RNRMOte5NEfJbmIAkwV6OUSfGW6MQeU2nASwp7+qa7HRW5CmoMiLQggtbuxsgRd5Qk5O9ZQgzjL13g2FHz40w/FhoUZYBxZLz55sDLw6Qqwhax8duyVaEKoy50elOBmwPK3hujCKhw0v+lnHUyAHB2lH+gHasDyOEtXfnsVYmF+oOTAv4HtQaQb9Vb6mxDWSZStCOqMW5gzGPawe/DsWoizKQ2VRoLSqkm1UTjYbUSizUzwnVyBOVPHpDj6F5vMhHUtMrd+cmN2pEjDaFR4YIXDb4T64Q7RiIyT9BRqVGD9F4OKDzvfFnlDrvpPJyStlSEoImALn3YQ2YI1XdiejeGjN3EKV5BYhJlci45Fo0MzcYHBea1Yqp3/xBOp4TBplmcw0C24eq6YCCUHTZdEhASU0TZ0lA5FzNICcq4pXUkOR8h2JyxXzweti0AYSwcIUVrOuJxyYquRI5TfL24LieAgSZCO0Qtid+osmSnhplIDFZxeEevC3lq/TJwtKfHXYbP0tSZUS6MLtOixR7atZQVlONK/TgG6SAvgytl6PxXsjHcqaMrtf/AN+FiWA=) to confirm this issue.

```vue
<script setup >
//           ↑ should be reported
</script>
```

**Comment by FloEdelmann** [comment]
Interestingly, both rules actually already work for the top-level `<template>` tag, but not for other top-level tags.

I'd say that this is a bug and it shouldn't be an option but a new default behavior. A new major release is due soon anyway (see #1850), so even if this is considered a breaking change, it would be a good time.

@ota-meshi, what do you thing?

**Comment by ota-meshi** [comment]
I think it's good to change those rules and apply the rules to the top level tags 👍
